### PR TITLE
parser: support interval {div/mult} float

### PR DESCRIPTION
**Pull request by danhhz**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/14614)

### CODE REVIEW

**Comment by danhhz** [comment]
@dt ping?

File: pkg/util/duration/duration.go [hunk 1]
```diff
@@ -154,6 +154,16 @@ func (d Duration) Div(x int64) Duration {
 	return Duration{d.Months / x, d.Days / x, d.Nanos / x}
 }
 
+// MulFloat returns a Duration representing a time length of d*x.
+func (d Duration) MulFloat(x float64) Duration {
+	return Duration{int64(float64(d.Months) * x), int64(float64(d.Days) * x), int64(float64(d.Nanos) * x)}
```
Comment (dt) [inline pkg/util/duration/duration.go]: long lines?

**Review by dt — APPROVED** [review]

**Comment by danhhz** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/util/duration/duration.go, line 159 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14614#-KguZ3cz74Q58G1pgVGI-r1-159:-KhNGWWXcQkXeDCpLssG:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac02ebd45b1d29617223da91d53796fd3d11e68f/pkg/util/duration/duration.go#L159)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

long lines?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14614)*

**Comment by dt** [comment]
Reviewed 1 of 3 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14614)*

**Pull request merged**